tests/unit/src/formatMarkup.test.js

```diff
@@ -600,4 +600,56 @@ describe('Format markup', () => {
         `);
     });
   });
+
+  describe('Complex', () => {
+    let MyComponent = {
+      template: `
+      <div id="header" data-server-rendered>
+        <!--v-if-->
+        <label data-test="input" data-v-1ae75a9f="">
+          Void and Attributes per line Example:
+          <input />
+          <input type="range" />
+          <input type="range" max="50" />
+          <input type="range" max="50" id="slider" >
+        </label>
+        <p class="">Empty attribute example</p>
+        <div></div>
+        <ul><li><a href="#">Link text on same line</a></li></ul>
+      </div>`
+    };
+    test('Two Complex', async () => {
+      const wrapper = mount(MyComponent);
```

Comment:
So one problem I've found is that
```js
console.log(wrapper.html());
```
outputs
```html
<div id="header" data-server-rendered="">
  <!--v-if--><label data-test="input" data-v-1ae75a9f=""> Void and Attributes per line Example: <input><input type="range"><input type="range" max="50"><input type="range" max="50" id="slider"></label>
  <p class="">Empty attribute example</p>
  <div></div>
  <ul>
    <li><a href="#">Link text on same line</a></li>
  </ul>
</div>
```
Note that the template was defined with returns and indentation after every `<input />`, but the built-in `.html()` serializer is removing that information.

* Short term option:
  * Change `vueSnapshots.formatting.tagsWithWhitespacePreservered` to ONLY accept an array (and for falsey values to produce the default `['a', 'pre']`).
  * This removes the need to support the `true` option for now.
* Long term option:
  * If the user passed in a VTU wrapper, we could inspect it to find the original Vue template definition and manually parse that to map the intended indentation to the snapshot. This would likely be a very high amount of effort for fairly little value. But there is still some value.
  * If an existing Vue template AST library existed, it might make this more feasible.

Reply:
Technically, we could fix this for non-vue-wrapper cases. Where you are passing in an arbitrary string of HTML. Then the original intended indentation would be kept. But I don't really care about that scenario for this library. If the `diffable` formatter was extracted to its own library, that would make more sense.

I created #60 to apply the short term option.

Reply:
I merged in #60. Thanks for putting the time into this (and the other PRs). I'm going to close this PR for now, unless you have any other ideas.